Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -435,22 +437,25 @@ class BrowserViewController: UIViewController {
     }
     
     fileprivate func setupTabs() {
-        let isPrivate = Preferences.Privacy.privateBrowsingOnly.value
-        let noTabsAdded = tabManager.tabsForCurrentMode.isEmpty
-        
-        var tabToSelect: Tab?
-        
-        if noTabsAdded {
-            // Two scenarios if there are no tabs in tabmanager:
-            // 1. We have not restored tabs yet, attempt to restore or make a new tab if there is nothing.
-            // 2. We are in private browsing mode and need to add a new private tab.
-            tabToSelect = isPrivate ? tabManager.addTab(isPrivate: true) : tabManager.restoreAllTabs()
-        } else {
-            tabToSelect = tabManager.tabsForCurrentMode.last
-        }
-        
         contentBlockListDeferred?.uponQueue(.main) { _ in
-            self.tabManager.selectTab(tabToSelect)
+            let isPrivate = Preferences.Privacy.privateBrowsingOnly.value
+            let noTabsAdded = self.tabManager.tabsForCurrentMode.isEmpty
+            
+            var tabToSelect: Tab?
+            
+            if noTabsAdded {
```

Comment:
This is a question, but should we even execute all of this code (lines 446-453) if the `loadQueue` has items to execute? All of those (currently anyway), lead to new tab creation, right?